### PR TITLE
verification-sdk: React component foundations

### PR DESCRIPTION
**Pull request by pgarciaegido**
## Description

Design doc: https://docs.google.com/document/d/1iIhjE_CZNA0F4E49mxFN53AADh87jfb2LoRMer_jup0/edit#heading=h.ivay2yrvkmfi
API Design Doc: https://docs.google.com/document/d/1s5EFngSxkcieS3yUPUhm-nIE8bOb1xdfB--sCsmJJ2s/edit

Basically declaring interfaces and placing the base component. Lets discuss if this is the appropriate place for them, and also the interface of the SDK in general.

## Test plan

Build works as expected. Not much to test for now.

### CODE REVIEW

File: packages/client/base/src/types/hosted/index.ts [hunk 1]
```diff
@@ -46,5 +48,39 @@ export type CheckoutProps = {
...
+        twitter?: string;
+        discord?: string;
+    };
+};
+
+export const VerificationProducts = {
+    PAYMENTS_CREDIT_CARD: "payments:credit-card",
+    PAYMENTS_CROSS_CHAIN: "payments:cross-chain",
+};
+export type VerificationProducts = ObjectValues<typeof VerificationProducts>;
```
Comment (angel-paella) [inline packages/client/base/src/types/hosted/index.ts:79]: not sure if I'd call it products here, but not sure of a better name. VerificationTargets maybe?

Products I thought it was Items as in NFTs

Comment (angel-paella) [reply packages/client/base/src/types/hosted/index.ts:79]: VerificationScopes probably better

File: packages/client/base/src/types/hosted/index.ts [hunk 2]
```diff
@@ -46,5 +48,39 @@ export type CheckoutProps = {
...
+
+export type CrossmintVerificationButtonProps = {
+    collectionId: string;
+    products: VerificationProducts[];
+    fields: CrossmintVerificationFields;
```
Comment (angel-paella) [inline packages/client/base/src/types/hosted/index.ts:84]: Either use CrossmintVerification as a preffix or not, but consistently XD

I would remove the Crossmint preffix

File: packages/client/ui/react-ui/src/components/hosted/CrossmintVerificationButton.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { CSSProperties, MouseEvent } from "react";
+
+import { CrossmintVerificationButtonProps } from "@crossmint/client-sdk-base";
+
+export type CrossmintVerificationButtonReactProps = CrossmintVerificationButtonProps & {
```
Comment (angel-paella) [inline packages/client/ui/react-ui/src/components/hosted/CrossmintVerificationButton.tsx:5]: do we use CrossmintPaymentsButton or something? Why the Crossmint preffix?

File: packages/client/base/src/types/hosted/index.ts [hunk 4]
```diff
@@ -46,5 +48,39 @@ export type CheckoutProps = {
     display?: "same-tab" | "new-tab" | "popup";
     delivery?: "custodial" | "non-custodial" | "all";
     paymentMethods?: PaymentMethod[];
-}
+};
 export type SigninMethods = "metamask" | "solana";
+
+export const VerificationCollectionType = {
```
Comment (mPaella) [inline packages/client/base/src/types/hosted/index.ts:54]: This stuff should be in its own file, this is meant for hosted checkout types

File: packages/client/base/src/types/hosted/index.ts (hunk 1 above)
Comment (pgarciaegido) [reply packages/client/base/src/types/hosted/index.ts:79]: We got to discuss what approach to take designing this interface.

Approach 1: the SDK user to pass us the flows they want to perform (kyc, collection verification)
Approach 2: the SDK user sending us for what product / crossmint offering the need verification for (credit card payments, cross-chain payments)

I think we have much more control with Approach 2. Now the question is, how do we call "payments:credit-card", "payments:cross-chain". Product? Flow? Scope?

Just adding this comment for reference. I think scope makes sense.

File: packages/client/ui/react-ui/src/components/hosted/CrossmintVerificationButton.tsx (hunk 3 above)
Comment (pgarciaegido) [reply packages/client/ui/react-ui/src/components/hosted/CrossmintVerificationButton.tsx:5]: Yes, the name of our SDK is CrossmintPayButton

Screenshot 2024-03-05 at 18 41 11

I don't know why the Crossmint prefix. But some other products use it as well. Example with Persona:
Screenshot 2024-03-05 at 18 41 54

File: packages/client/base/src/types/hosted/verification.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        website?: string;
+        twitter?: string;
+        discord?: string;
+    };
+};
+
+export const VerificationScopes = {
+    PAYMENTS_CREDIT_CARD: "payments:credit-card",
+    PAYMENTS_CROSS_CHAIN: "payments:cross-chain",
+};
```
Comment (mPaella) [inline packages/client/base/src/types/hosted/verification.ts:27]: As const

File: packages/client/base/src/types/index.ts [hunk 6]
```diff
@@ -1,6 +1,7 @@
 export * from "./errors";
 export * from "./events";
 export * from "./hosted";
+export * from "./hosted/verification";
```
Comment (mPaella) [inline packages/client/base/src/types/index.ts:4]: Typically this export should go in /hosted/index.ts

So just export * from ./verification

**Review by mPaella — APPROVED** [review]
Approved with nits

**Pull request merged**